### PR TITLE
Plugins: Add CDN support to sandboxed frontend plugins

### LINKED ISSUE #69846: Fix plugins with code coming from a CDN
Plugin running inside the sandbox should work with CDNs the same as non-sandboxed plugins do. 

To fix:

* Image loading
* sourcemaps should load

### PR DESCRIPTION
**Pull request by academo**
**What is this feature?**

Adds CDN support for plugins loaded inside the frontend sandbox

**Why do we need this feature?**

For plugins that should load from a CDN work inside the sandbox

**Who is this feature for?**

Plugins running in grafana

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/grafana/issues/69846

**Special notes for your reviewer:**

Please check that:

- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: public/app/features/plugins/cdn/utils.test.ts [hunk 1]
```diff
@@ -1,29 +1,19 @@
...
-        format: 'amd',
-        loader: 'cdn-loader',
-        encapsulateGlobal: false,
-        cjsRequireDetection: true,
-        cjsDeferDepsExecute: false,
-        esModule: true,
-        authorization: false,
-      },
-    };
+import { extractPluginIdVersionFromUrl, transformPluginSourceForCDN } from './utils';
```
Comment (academo) [inline public/app/features/plugins/cdn/utils.test.ts:3]: This is mostly the same tests that used to be the systemjs cdn plugin tests.

**Review by jackw — APPROVED** [review]
Tested locally against the agenty-flowcharting-panel and working well. Great stuff! 🚀

**Comment by xnyo** [comment]
I can't get it to work with the CDN :disappointed: but maybe I am doing something wrong.

I used this config:

```
[feature_toggles]
enable = pluginsFrontendSandbox

[plugins]
cdn_base_url = https://grafana-assets.grafana.net/plugins

[plugin.grafana-clock-panel]
cdn = true
cdn_version = 2.1.3

[plugin.agenty-flowcharting-panel]
cdn = true
cdn_version = 0.9.1
```

Clock panel works fine.

Flowcharting (which also seems to be Angular) however, doesn't load properly when the sandbox is on. It looks like the path for module.js is constructed correctly, but the one for additional files is not:

For example, when I select the plugin for a panel, the browser tries to fetch this file:
https://grafana-assets.grafana.net/plugins/agenty-flowcharting-panel/public/public/plugins/agenty-flowcharting-panel/partials//flowchart/index.html

Which doesn't exist because the path is wrong (`public` is present twice and there is no version).

The correct URL should be:
https://grafana-assets.grafana.net/plugins/agenty-flowcharting-panel/0.9.1/public/plugins/agenty-flowcharting-panel/partials//flowchart/index.html

When I turn off the sandbox feature flag, the plugin works fine

**Comment by jackw** [comment]
Ah sorry for the confusion @xnyo - the sandbox doesn't work with Angular panels. My comment above related to the agenty flowchart panel was me testing without the sandbox active.

**Comment by xnyo** [comment]
Oh okay! Thanks for clarifying @jackw :pray:  sounds good!

**Pull request merged**